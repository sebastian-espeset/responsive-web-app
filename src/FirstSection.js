import { React, useState, useEffect } from 'react'
import { StyledFirstSection,
        StyledHeadline,
        StyledVideoContainer
      } from './styles/FirstSection.style'
import axios from 'axios';
//Familar Pro Light font
export default function FirstSection() {
  const [videoState, setVideoState]=useState({
    videoLink:""
  });
  useEffect(() => {
    axios.get("https://vimeo.com/api/oembed.json?url=https://vimeo.com/327864855")
        .then(res => {
          const parser = new DOMParser();
          let videoLinkSource = parser.parseFromString(res.data.html,'text/html')
            setVideoState({
              videoLink:videoLinkSource.body.innerHTML
            });
        })
},[]);
  return (
    <StyledFirstSection>
      <StyledHeadline data-testid='first-section-headline-test-id'>
        <p>Dental benefits that reward you for brushing your teeth.</p>
      </StyledHeadline>
      <StyledVideoContainer>
      {videoState.videoLink.length > 1 ? <div dangerouslySetInnerHTML={{__html:videoState.videoLink}}/> : <div>Video loading...</div>}
      </StyledVideoContainer>
    </StyledFirstSection>
  )
}


