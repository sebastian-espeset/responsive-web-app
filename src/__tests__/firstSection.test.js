import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import FirstSection from '../FirstSection';

afterEach(()=>{
    cleanup();
});

test('First sections headline should render',()=>{
    //Arrange
    render(<FirstSection/>)
    //Act
    const firstSectionHeadline = screen.getByTestId('first-section-headline-test-id');
    //Assert
    expect(firstSectionHeadline).toBeInTheDocument();
});

// test will create a 'snapshot' of the html file, and then compare it to potential changes made    
test('first section component to match snapshot',()=>{
    const tree = renderer.create(<FirstSection/>).toJSON();
    expect(tree).toMatchSnapshot();
})