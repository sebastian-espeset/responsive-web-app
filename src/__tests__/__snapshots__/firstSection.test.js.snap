// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first section component to match snapshot 1`] = `
<div
  className="sc-bczRLJ iuoXcI"
>
  <div
    className="sc-gsnTZi crTZHj"
    data-testid="first-section-headline-test-id"
  >
    <p>
      Dental benefits that reward you for brushing your teeth.
    </p>
  </div>
  <div
    className="sc-dkzDqf biTqlf"
  >
    <div>
      Video loading...
    </div>
  </div>
</div>
`;
